### PR TITLE
Add Performance doc to Java

### LINKED ISSUE #3153: Add performance optimization docs
Splunk has a decent doc on how to evaluate the performance overhead of the OTel Java Agent:
https://docs.splunk.com/Observability/gdi/get-data-in/application/java/performance.html

I think it would be great if OTel had similar content on our doc site. Jason suggested Splunk could possibly contribute their content. 

cc @theletterf @breedx-splk

### PR DESCRIPTION
**Pull request by theletterf**
Fixes #3153.

We've got legal approval to upstream the original doc.

### CODE REVIEW

**Comment by theletterf** [comment]
I think the doc could be adapted to other languages with some changes.

File: content/en/docs/languages/java/performance.md [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+overhead. For example, using `@WithSpan` on every method results in a high span
+volume, which in turn increases noise in the data and consumes more system
+resources.
+
+### Provision adequate resources
+
+Make sure to provision enough resources for your instrumentation and for the
+Collector. The amount of resources such as memory or disk depend on your
+application architecture and needs. For example, a common setup is to run the
+instrumented application on the same host as the Distribution of OpenTelemetry
```
Comment (tiffany76) [inline content/en/docs/languages/java/performance.md:73]:
```suggestion
instrumented application on the same host as the distribution of OpenTelemetry
```

File: content/en/docs/languages/java/performance.md [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+
+Different users of your application or service might notice different aspects of
+agent overhead. For example, while end users might notice degradation in service
+latency, power users with heavy workloads pay more attention to CPU overhead. On
+the other hand, users who deploy frequently, for example due to elastic
+workloads, care more about startup time.
+
+Reduce your measurements to factors that are sure to impact the user experience
+of your application, so as not to produce datasets that contain irrelevant
+information. Some examples of measurements include the following:
```
Comment (tiffany76) [inline content/en/docs/languages/java/performance.md:133]:
```suggestion
Reduce your measurements to factors that are sure to impact user experience, 
so your datasets don't contain irrelevant
information. Some examples of measurements include the following:
```

File: content/en/docs/languages/java/performance.md [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+- Total memory allocated and maximum heap used
+- Garbage collection pause time
+- Startup time in milliseconds
+- Average and percentile 95 (p95) service latency
+- Network read and write average throughput
+
+### Prepare a suitable test environment
+
+By measuring agent overhead in a controlled test environment you can better
+control and identify the factors affecting performance. When preparing a test
```
Comment (tiffany76) [inline content/en/docs/languages/java/performance.md:145]:
```suggestion
By measuring agent overhead in a controlled test environment you can better
identify the factors affecting performance. When preparing a test
```

File: content/en/docs/languages/java/performance.md [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+For Java applications, use a warm-up phase prior to starting measurements. The
+JVM is a highly dynamic machine that performs a large number of optimizations
+through just-in-time compilation (JIT). The warm-up phase helps the application
+to finish most of its class loading and gives the JIT compiler time to run the
+majority of optimizations.
+
+Make sure to run a large number of requests and to repeat the test pass many
+times. This repetition helps to ensure a representative data sample. Include
+error scenarios in your test data. Simulate an error rate similar to that of a
+normal workload, typically between 2% to 10%.
```
Comment (tiffany76) [inline content/en/docs/languages/java/performance.md:171]:
```suggestion
normal workload, typically between 2% and 10%.
```

File: content/en/docs/languages/java/performance.md [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+
+To identify which factors might be affecting performance and causing agent
+overhead, collect measurements in the same environment after modifying a single
+factor or condition.
+
+### Analyze the agent overhead data
+
+After collecting data from multiple passes, you can compare averages using
+simple statistical tests to check for significant differences, or plot results
+in a chart.
```
Comment (tiffany76) [inline content/en/docs/languages/java/performance.md:183]:
```suggestion
After collecting data from multiple passes, you can plot results
in a chart or compare averages using statistical tests to check for significant differences.
```

**Comment by theletterf** [comment]
Excellent review, thanks @tiffany76 !

File: content/en/docs/languages/java/performance.md [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+For example, the following option turns off the JDBC instrumentation:
+`-DOpenTelemetry.instrumentation.jdbc.enabled=false`
+
+### Allocate more memory for the application
+
+Increasing the maximum heap size of the JVM using the `--Xmx<size>` option might
+help in alleviating agent overhead issues, as instrumentations can generate a
+large number of short-lived objects in memory.
+
+### Reduce manual instrumentation to a minimum
```
Comment (cartermp) [inline content/en/docs/languages/java/performance.md:61]:
```suggestion
### Reduce manual instrumentation to what you need
```

File: content/en/docs/languages/java/performance.md [hunk 7]
```diff
@@ -0,0 +1,186 @@
...
+
+Manual instrumentation might introduce inefficiencies that increase agent
+overhead. For example, using `@WithSpan` on every method results in a high span
+volume, which in turn increases noise in the data and consumes more system
```
Comment (cartermp) [inline content/en/docs/languages/java/performance.md:65]:
```suggestion
Too much manual instrumentation might introduce inefficiencies that increase agent
overhead. For example, using `@WithSpan` on every method results in a high span
volume, which in turn increases noise in the data and consumes more system
```

**Review by cartermp — APPROVED** [review]

File: content/en/docs/languages/java/performance.md [hunk 8]
```diff
@@ -0,0 +1,186 @@
...
+Make sure to provision enough resources for your instrumentation and for the
+Collector. The amount of resources such as memory or disk depend on your
+application architecture and needs. For example, a common setup is to run the
+instrumented application on the same host as the OpenTelemetry Collector. In
+that case, consider rightsizing the resources for the Collector and optimize its
+settings. See [Scaling](/docs/collector/scaling/)
+
+## Constraints impacting the performance of the Java agent
+
+In general, the more telemetry you collect from your application, the bigger is
```
Comment (jaydeluca) [inline content/en/docs/languages/java/performance.md:79]:
```suggestion
In general, the more telemetry you collect from your application, the greater the
```

File: content/en/docs/languages/java/performance.md [hunk 9]
```diff
@@ -0,0 +1,186 @@
...
+
+1.  Make sure that the configuration of the test environment resembles
+    production.
+2.  Isolate the application under test from other services that might interfere.
+3.  Turn off or remove all unnecessary system services on the application host.
+4.  Ensure that the application has enough system resources to handle the test
+    workload.
+
+### Create a battery of realistic tests
+
```
Comment (jaydeluca) [inline content/en/docs/languages/java/performance.md:156]: just a thought but would it make sense to provide a general disclaimer/warning about being careful about running up bills with observability backends when designing and running these tests? For example a large scale load test to determine the overhead but with a sample rate of 100%, might be fine for some companies but could also provide a shock bill if they have a tight budget and aren't prepared.

File: content/en/docs/languages/java/performance.md [hunk 10]
```diff
@@ -0,0 +1,186 @@
...
+The volume of spans processed by the instrumentation might impact agent
+overhead. You can configure trace sampling to adjust the span volume and reduce
+resource usage. See [Sampling](/docs/languages/java/sampling).
+
+### Turn off specific instrumentations
+
+You can further reduce agent overhead by turning off instrumentations that
+aren't needed or are producing too many spans. To turn off an instrumentation,
+use `-DOpenTelemetry.instrumentation.<name>.enabled=false` or the
```
Comment (trask) [inline content/en/docs/languages/java/performance.md:48]:
```suggestion
use `-Dotel.instrumentation.<name>.enabled=false` or the
```

File: content/en/docs/languages/java/performance.md [hunk 11]
```diff
@@ -0,0 +1,186 @@
...
+`OpenTelemetry_INSTRUMENTATION_<NAME>_ENABLED` environment variable, where
```
Comment (trask) [inline content/en/docs/languages/java/performance.md:49]:
```suggestion
`OTEL_INSTRUMENTATION_<NAME>_ENABLED` environment variable, where
```

File: content/en/docs/languages/java/performance.md [hunk 12]
```diff
@@ -0,0 +1,186 @@
...
+`<name>` is the name of the instrumentation.
+
+For example, the following option turns off the JDBC instrumentation:
+`-DOpenTelemetry.instrumentation.jdbc.enabled=false`
```
Comment (trask) [inline content/en/docs/languages/java/performance.md:53]:
```suggestion
`-Dotel.instrumentation.jdbc.enabled=false`
```

File: content/en/docs/languages/java/performance.md [hunk 13]
```diff
@@ -0,0 +1,186 @@
...
+
+### Allocate more memory for the application
+
+Increasing the maximum heap size of the JVM using the `--Xmx<size>` option might
```
Comment (trask) [inline content/en/docs/languages/java/performance.md:57]:
```suggestion
Increasing the maximum heap size of the JVM using the `-Xmx<size>` option might
```

File: content/en/docs/languages/java/performance.md (hunk 9 above)
Comment (theletterf) [reply content/en/docs/languages/java/performance.md:156]: Good point! I'll add that.

Comment (theletterf) [reply content/en/docs/languages/java/performance.md:156]: How about this wording?

```
{{% alert title="Note" color="info" %}}

Tests might increase costs when targeting observability backends and other
commercial services. Plan your tests accordingly or consider using alternative
solutions, such as self-hosted or locally run backends.

{{% /alert %}}
```

**Review by jaydeluca — APPROVED** [review]

**Comment by svrnm** [comment]
@theletterf can this moved from draft state?

**Comment by theletterf** [comment]
Soon. Providing a courtesy last check slot to Legal here.

**Comment by theletterf** [comment]
Message from Legal:

> Thank you for your patience! This has been approved. It should be good to go now.

Publishing!
